src/models/point/model_point.py

```diff
@@ -73,10 +90,67 @@ def get_model_event_type(self) -> EventType:
     def update(self, **kwargs):
         super().update(**kwargs)
 
-        point_store = PointStoreModel.find_by_point_uuid(self.uuid)
-        updated = point_store.update(self, 0)
+        point_store: PointStoreModel = PointStoreModel.find_by_point_uuid(self.uuid)
+        updated = point_store.update(0)
         self.point_store = point_store
         if updated:
-            point_store.publish_cov(self)
+            self.publish_cov(self.point_store)
 
         return self
+
+    @classmethod
+    def apply_offset(cls, original_value: float, value_offset: float, value_operation: MathOperation) -> float or None:
+        """Do calculations on original value with the help of point details"""
+        if original_value is None or value_operation is None:
+            return original_value
+        value = original_value
+        if value_operation == MathOperation.ADD:
+            value += value_offset
+        elif value_operation == MathOperation.SUBTRACT:
+            value -= value_offset
+        elif value_operation == MathOperation.MULTIPLY:
+            value *= value_offset
+        elif value_operation == MathOperation.DIVIDE:
+            value /= value_offset
+        elif value_operation == MathOperation.BOOL_INVERT:
+            value = not bool(value)
+        return value
+
+    @classmethod
+    def apply_scale(cls, value: float, input_min: float, input_max: float, output_min: float, output_max: float) \
+            -> float or None:
+        if value is None or input_min is None or input_max is None or output_min is None or output_max is None:
+            return value
+        value = ((value - input_min) * (output_max - output_min)) / (input_max - input_min) + output_min
+        return value
+
+    def publish_cov(self, point_store: PointStoreModel, device: DeviceModel = None, network: NetworkModel = None,
+                    service_name: str = None):
+        if point_store is None:
+            raise Exception('Point.publish_cov point_store cannot be None')
+        if device is None:
+            device = DeviceModel.find_by_uuid(self.device_uuid)
+        if network is None:
+            network = NetworkModel.find_by_uuid(device.network_uuid)
+        if device is None or network is None:
+            raise Exception(f'Cannot find network or device for point {self.point_uuid}')
```

Comment:
`raise Exception(f'Cannot find network or device for point {self.uuid}')`

There is no `point_uuid` on this model.